### ISSUE TITLE
Composition ignores incorrect `_Entity` definitions, but maybe it shouldn't

### ISSUE THREAD

**Issue by pcmanus** [apollographql/federation#2629]
Suppose 2 subgraphs defined as:
- subgraph `S1`:
    ```graphql
      type Query {
        t1: T1
        t2: T2
      }

      type T1 @key(fields: "id") {
        id: ID!
      }

      type T2 @key(fields: "id") {
        id: ID!
      }
    ```
- subgraph `S2`:
    ```graphql
      union _Entity = T1

      type T1 @key(fields: "id") {
        id: ID!
        v1: Int
      }

      type T2 @key(fields: "id") {
        id: ID!
        v2: Int
      }
    ```

Note that `S2` declares the (federation specific) `_Entity` union, but that it does so incorrectly: that union, by definition of the spec, should contains every entity type of the subgraph, that is every type defined with a `@key`, so it should include `T1` and `T2` in that case, but it only include `T1`.

Note that in practice you shouldn't really define `_Entity` manually like above and instead rely on your federation-compatible client library to generate it for you, but such `S2` could be the result of either a buggy federation client library, or of `_Entity` ending up hardcoded in the subgraph for some reason and not being properly updated afterwards.

Anyway, the behaviour in that case is that composition will "ignore" this and the query planner will later not know that the subgraph has this "issue", and will happily consider that it can ask `S2` to resolve the `T2` entity.

Very concretely, given a query like:
```graphql
query {
  t2 {
    v2
  }
}
```
the query planner will send to `T2` a query looking like:
```graphql
query($representation: [_Any!]!) {
  _entities(representation: $representation) {
    ... on T2 {
      v2
    }
  }
}
```

But technically, this last query is kind of invalid for that subgraph and validating it against the original schema above would give something like `GraphQLError: Fragment cannot be spread here as objects of type "_Entity" can never be of type "T2".`.

The problem is that the subgraph does not correctly implement the federation spec, but we should probably detect this pre-composition and return an error, rather than accepting it and generating subgraph queries that kind of don't validate against the subgraph schema.

I'll note that this problem is not new at all: while I haven't concretely checked, I'm almost certain all version of federation 2 have behaved this way and I'm 92.7% sure federation 1 did too. And I'll also note that it's possible that there exists subgraph looking like `S2` above, but that just "work" still in practice, because the same way that composition ignores the faulty `_Entity` definition, some federation client library might as well (I believe `@apollo/subgraph` does for instance).

Long story short, my own opinion is that it's probably worth tightening the validation on this, but I also suggest doing so only in an upcoming `2.5` release and not in any current branch, because this could technically break composition for someone that has no problem at the moment and that would be particularly unexpected in a patch release.
